scipy/stats/_hypotests.py

```diff
@@ -130,3 +133,224 @@ def epps_singleton_2samp(x, y, t=(0.4, 0.8)):
+        # therefore, we need to add it here
+        y[sup] = _cdf_cvm_inf(x[sup]) * (1 + 1./(12*n)) + _psi1_mod(x[sup]) / n
+        y[x >= n/3] = 1
+
+    if y.ndim == 0:
+        return y[()]
+    return y
+
+
+def cvm_test(rvs, cdf, args=()):
+    """
+    Perform the Cramer-von Mises test for goodness of fit.
+
+    This performs a test of the goodness of fit of a cumulative distribution
+    function F compared to the empirical distribution function F_n of
+    observed random variates X_1, ..., X_n that are assumed to be independent
+    and identically distributed ([1]_). The null hypothesis is that the X_i
+    have cumulative distribution F.
+
+    Parameters
+    ----------
+    rvs : array_like
+        The observed values of the random variables, it should be a 1-D array.
+    cdf : str or callable
+        If a string, it should be the name of a distribution in `scipy.stats`.
+        If a callable, that callable is used to calculate the cdf.
+        ``cdf(x, *args) -> float``, it should be a cdf.
+    args : tuple, optional
+        Distribution parameters. These are assumed to be known, see Notes.
+
+    Returns
+    -------
+    statistic : float
+        Cramer-von Mises statistic.
+    pvalue :  float
+        The p-value.
+
+    See Also
+    --------
+    kstest
+
+    Notes
+    -----
+    The p-value relies on the approximation given by equation 1.8 in [2]_.
+    It is important to keep in mind that the p-value is only accurate if
+    one tests a simple hypothesis, i.e. the parameters of the reference
+    distribution are known. If the parameters are estimated from the data
+    (composite hypothesis), the computed p-value is not reliable.
+
+    References
+    ----------
+    .. [1] https://en.wikipedia.org/wiki/Cramer-von_Mises_criterion
+    .. [2] Csorgo, S. and Faraway, J. (1996). The Exact and Asymptotic
+           Distribution of Cramer-von Mises Statistics. Journal of the
+           Royal Statistical Society, pp. 221-234.
+
+    Examples
+    --------
+    >>> from scipy import stats
+
+    Test the assumption of normality. In this example, one accepts
+    the assumption of a normal distribution at the significance level
+    of alpha=0.05.
+
+    >>> np.random.seed(626)
+    >>> x = stats.norm.rvs(size=500)
+    >>> w, p = stats.cvm_test(x, 'norm')
+    >>> w, p
+    (0.06342154705518796, 0.792680516270629)
+
+    We can also use `args` to check whether the data is drawn from a
+    shifted normal distribution, assuming a shift of 2 when the
+    true shift is 2.1. In that case, the null hypothesis is rejected
+    at the significance level of alpha=0.05.
+
+    >>> y = x + 2.1
+    >>> w, p = stats.cvm_test(y, 'norm', args=(2,))
+    >>> w, p
+    (0.4798693195559657, 0.044782228803623814)
+
+    Using a callable argument for cdf, this is equivalent to the
+    following:
+
+    >>> cdf = stats.norm.cdf
+    >>> stats.cvm_test(y, cdf, args=(2,))
```

Comment:
I notice the refguide build is failing
```
File "build/testenv/lib/python3.6/site-packages/scipy/stats/_hypotests.py", line 328, in cvm_test
Failed example:
    stats.cvm_test(y, cdf, args=(2,))
Expected nothing
Got:
    cvm_testResult(statistic=0.4798693195559657, pvalue=0.04478222880362348)
```

Is there some output missing on line 329?

Reply:
yes, i will add the output once there are further review comments that I need to address